sources/osgStats/TextGenerator.js

```diff
@@ -0,0 +1,181 @@
+        var uvs = buffer._uvs;
+
+        buffer._nbCharacters += 1;
+        var i = nbCharacters * 4 * 3;
+
+        vertexes[i++] = currentPosition[0];
+        vertexes[i++] = currentPosition[1];
+        vertexes[i++] = currentPosition[2];
+
+        var x = currentPosition[0] + this._characterWidth * sizeFactor;
+        var y = currentPosition[1] + this._characterHeight * sizeFactor;
+
+        vertexes[i++] = currentPosition[0];
+        vertexes[i++] = y;
+        vertexes[i++] = currentPosition[2];
+
+        vertexes[i++] = x;
+        vertexes[i++] = y;
+        vertexes[i++] = currentPosition[2];
+
+        vertexes[i++] = x;
+        vertexes[i++] = currentPosition[1];
+        vertexes[i++] = currentPosition[2];
+
+        var characterUV = this.getCharacterUV(characterCode);
+        i = nbCharacters * 4 * 2;
+
+        uvs[i++] = characterUV;
+        uvs[i++] = 0.0;
+
+        uvs[i++] = characterUV;
+        uvs[i++] = 1.0;
+
+        var u = characterUV + this._characterSizeUV;
+
+        uvs[i++] = u;
+        uvs[i++] = 1.0;
+
+        uvs[i++] = u;
+        uvs[i++] = 0.0;
+    },
+    generateText: function(textCursor, text, buffer, sizeFactor) {
+        var size = text.length;
+        if (buffer._nbCharacters + size >= buffer._maxCharacters) {
+            buffer.resize(buffer._maxCharacters * 2);
+            console.log('resize buffer to ' + buffer._maxCharacters * 2);
+        }
+        for (var i = 0; i < size; i++) {
+            var characterCode = text.charCodeAt(i);
+            this.generateCharacter(textCursor, characterCode, buffer, sizeFactor);
+            textCursor[0] += this._characterWidth * sizeFactor;
+        }
+        return size * (this._characterWidth * sizeFactor);
+    },
+    _fillCanvas: function(fontFamily, fontSize) {
+        // get font height
+        var size = fontSize + 'px';
+        var style = 'font-family: ' + fontFamily + '; font-size: ' + size + ';';
+        var pixelHeight = determineFontHeight(style);
+        notify.debug(fontFamily + ' ' + size + ' ==> ' + pixelHeight + ' pixels high.');
+
+        this._characterHeight = Math.ceil(pixelHeight);
+        var font = fontSize + "px '" + fontFamily + "'";
+        var canvas = document.createElement('canvas');
+        var ctx = canvas.getContext('2d');
+        this._canvas = canvas;
+        ctx.font = font;
+        ctx.textBaseline = 'middle';
+
+        this._characterWidth = ctx.measureText(' ').width;
+        this._characterWidth = Math.ceil(this._characterWidth);
+
+        var nbCharacters = this._characterLastCode - this._characterFirstCode + 1;
+        notify.info(
+            'Stats character size ' +
+                this._characterWidth +
+                ' x ' +
+                this._characterHeight +
+                ' texture width ' +
+                this._characterWidth * nbCharacters
+        );
+
+        canvas.width = Math.ceil(this._characterWidth * nbCharacters);
```

Comment:
Ah yes! you are right mipmaps or clamping modes could be the unsupported NPOT stuff in webGL.